code/controllers/subsystem/job.dm

```diff
@@ -159,7 +179,7 @@ SUBSYSTEM_DEF(job)
 	JobDebug("Running FOC, Job: [job], Level: [level], Flag: [flag]")
 	var/list/candidates = list()
 	for(var/mob/dead/new_player/player in unassigned)
-		if(QDELETED(player) || is_banned_from(player.ckey, job.title))
+		if(QDELETED(player) || (is_banned_from(player.ckey, list(job.get_jkey(), job.get_jkey(FALSE)))))
```

Comment:
`list(job.get_jkey())`: returns JOB_KEY as default. Psychiatrist will return PSYCHI
`job.get_jkey(FALSE))`: returns JOB_KEY as parent type for gimmicks. Psychiatrist will return MEDICAL DOCTOR

Reply:
I find that excessively confusing, I would suggest either using named args, a different proc name, or a define instead of TRUE/FALSE like GIMMICKS_INCLUDED etc

Reply:
maybe double checking its origin wouldn't be necessary. as banning clown is expected you to be banned from clown only, not stage magician as well.